aquadoggo/migrations/20220509090252_create-operations.sql

```diff
@@ -0,0 +1,29 @@
+-- SPDX-License-Identifier: AGPL-3.0-or-later
+
+CREATE TABLE IF NOT EXISTS operations_v1 (
+    author              VARCHAR(64)       NOT NULL,
+    document_id         VARCHAR(64)       NOT NULL,
+    operation_id        VARCHAR(64)       NOT NULL UNIQUE,
+    entry_hash          VARCHAR(68)       NOT NULL UNIQUE,
+    action              VARCHAR(16)       NOT NULL,
+    schema_id           TEXT              NOT NULL,
+    previous_operations TEXT              NULL,
+    PRIMARY KEY (operation_id)
+);
+
+-- With the above "previous_operations" column in operations_v1 table we may no longer need this 
+-- relation table. Can we remove it or are there reasons it's useful?
+CREATE TABLE IF NOT EXISTS previous_operations_v1 (
+    parent_operation_id    VARCHAR(64)       NOT NULL,
+    child_operation_id     VARCHAR(64)       NOT NULL
+);
```

Comment:
I am trying out storing `previous_operations` as a string value in the `operations_v1` table. This has the benefit of being able to retrieve the previous operations for an operation without needing to perform any joins. We are already joining for the fields, so saving one here made sense. 

This makes the `previous_operations_v1` table slightly reduntant for now, although it may still be useful in the future. I'm happy to leave it in for now or remove it until it is needed.

(or maybe I'm missing some important use for it ;-p)